packages/react-ssr/scripts/build.ts

```diff
@@ -70,12 +71,47 @@ const createReactWrapperModules = async ({ entryPoints, distRoot }) => {
   console.info('writing modules completed!');
 };
 
-createReactWrapperModules({
-  entryPoints: readdirSync(COMPONENTS_DIR)
-    .filter((file) => file.startsWith('gcds') && file.endsWith('.js') && !file.endsWith('2.js'))
-    .map((file) => join(COMPONENTS_DIR, file)),
-  distRoot: join(DIST_DIR),
-}).catch((err) => {
-  console.error('unexpected error generating module!', err);
-  exit(1);
-});
+// Copy over the required stencil files, add required patch and modify imports in component files
+const patchStencil = async () => {
+
+  // Updating stencil import in component files
+  const componentOptions = {
+    files: './dist/esm/components/*.js',
+    from: '@stencil/core/internal/client',
+    to: '../lib/stencil',
+  };
+  
+  await replaceInFile(componentOptions);
+
+  // Patching stencil
+  // Will need to check if the from property needs to be updated when we update @stencil/core
+  const stencilOptions = {
+    files: './dist/esm/lib/stencil/index.js',
+    from: 'var win = typeof window !== "undefined" ? window : {};',
+    to: "var  win = typeof window !== 'undefined' ? window : globalThis || {};",
+  };
+
+  try {
+    const results = await replaceInFile(stencilOptions);
+
+    if (results[0].hasChanged) {
+      createReactWrapperModules({
+        entryPoints: readdirSync(COMPONENTS_DIR)
+          .filter((file) => file.startsWith('gcds') && file.endsWith('.js') && !file.endsWith('2.js'))
```

Comment:
What is '2.js' here?

Reply:
If you look in the `dist/esm/components` folder, some of our components get generated into to files (ex: `gcds-icon.js` and `gcds-icon2.js`). That extra bit on that statement stops the `createReactWrapperModules` from using the files that end in `2.js` to avoid a duplicate export error.

Reply:
Thanks! I'm curious why we have those generated files, but it doesn't seem to cause any issues at the moment. It'd be nice to have this written down as a comment explaining why it looks for the `2.js` but it's already in existing code so we can keep it as it is.